jfix-stdlib-concurrency/src/main/java/ru/fix/stdlib/concurrency/futures/PendingFutureLimiter.java

```diff
@@ -187,12 +185,16 @@ public PendingFutureLimiter changeThresholdFactor(float thresholdFactor) {
             awaitOpportunityToEnqueueAndPurge();
         }
 
-        counter.incrementAndGet();
-        pendingFutures.put(future, System.currentTimeMillis());
-        future.handleAsync((any, exc) -> {
+        CompletableFuture<T> resultFuture = future.handleAsync((any, exc) -> {
             if (exc != null) {
                 log.error(exc.getMessage(), exc);
+                throw new CompletionException(exc);
+            } else {
+                return any;
             }
+        });
+
+        resultFuture.handleAsync((any, exc) -> {
             long value = counter.decrementAndGet();
             pendingFutures.remove(future);
```

Comment:
Seems like we are going to end up with OOM: you are not removing resultFuture from pendingFutures

Reply:
Fixed